File: android/app/src/main/java/sol/vortexx/android/identity/api/SeedProvider.kt
package sol.vortexx.android.identity.api

/**
 * 24-word BIP39 mnemonic source. Split from [IdentityRepository] so that:
 *   - wave 6 only generates; wave 20 adds import / restore without changing
 *     consumers
 *   - tests can deterministically inject a known mnemonic
 */
interface SeedProvider {
    fun generate(): Mnemonic
    fun toSeed(mnemonic: Mnemonic, passphrase: CharArray = CharArray(0)): ByteArray
}

@JvmInline
value class Mnemonic(val words: List<String>) {
    init { require(words.size == 24) { "BIP39 mnemonic must be 24 words (got ${words.size})" } }
    override fun toString(): String = words.joinToString(" ")
}

File: ios/Modules/Sources/Identity/api/SeedProvider.swift
import Foundation

/// 24-word BIP39 mnemonic source.
///
/// Split from [IdentityRepository] so:
///   - Wave 6 only generates; Wave 20 adds import/restore without changing
///     consumers.
///   - Tests can deterministically inject a known mnemonic.
public protocol SeedProvider: Sendable {
    func generate() throws -> Mnemonic
    func toSeed(mnemonic: Mnemonic, passphrase: Data) throws -> Data
}

public extension SeedProvider {
    func toSeed(mnemonic: Mnemonic) throws -> Data { try toSeed(mnemonic: mnemonic, passphrase: Data()) }
}

public struct Mnemonic: Equatable, Sendable {
    public let words: [String]
    public init(words: [String]) throws {
        guard words.count == 24 else { throw SeedError.badWordCount(words.count) }
        self.words = words
    }
    public var phrase: String { words.joined(separator: " ") }
}

public enum SeedError: Error, Equatable {
    case badWordCount(Int)
    case wordlistUnavailable
    case unknownWord(String)
}
